qcodes/instrument_drivers/Keysight/keysight_e4980a.py

```diff
@@ -229,13 +229,23 @@ def __init__(self,
         self.add_parameter(
             "range",
             get_cmd=":FUNCtion:IMPedance:RANGe?",
-            set_cmd=":FUNCtion:IMPedance:RANGe {}",
+            set_cmd=self._set_range,
             unit='Ohm',
             vals=Enum(0.1, 1, 10, 100, 300, 1000, 3000, 10000, 30000, 100000),
             docstring="Selects the impedance measurement range, also turns "
                       "the auto range function OFF."
         )
 
+        self.add_parameter(
+            "imp_autorange_enabled",
+            get_cmd=":FUNCtion:IMPedance:RANGe:AUTO?",
+            set_cmd=":FUNCtion:IMPedance:RANGe:AUTO {}",
+            vals=Bool(),
+            val_mapping=create_on_off_val_mapping(on_val="1",
+                                                  off_val="0"),
+            docstring="Enables the auto-range for impedance measurement."
```

Comment:
Its not really needed to have a validator if you use a val_mapping since that automatically generates a validator to match all the valid values in the val_mapping. Currently the val_mapping generates a lot of valid values that are then being rejected by the validator